herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -1602,6 +1654,69 @@ public TableSpaceCheckpoint(
             this.tablesCheckpoints = tablesCheckpoints;
         }
     }
+    //this method returns a map with all scan values (record numbers , table digest,digestType, next autoincrement value, table name, tablespacename, query used for table scan )
+    public Map<String,Object> createAndWriteTableDigest(TableSpaceManager tableSpaceManager, String tableSpace,String tableName ) throws IOException{
```

Comment:
You'd better create a TableChecksum class that contains all of the returned fields.

A Map is untyped, no structure, it will be harder to modify in the future.

Remember to add getters/setters for each field, in order to make json serialization work properly (Netbeans has a utility function to create getters/setters)